src/internal/pjs/api_server.go

```diff
@@ -306,3 +298,40 @@ func toJobInfo(job pjsdb.Job) (*pjs.JobInfo, error) {
 	}
 	return jobInfo, nil
 }
+
+func (a *apiServer) resolveJob(ctx context.Context, jobCtx string, jobID int64) (id pjsdb.JobID, err error) {
+	// handle job context and request validation.
+	if jobCtx != "" {
+		id, err = a.resolveJobCtx(ctx, jobCtx)
+		if err != nil {
+			return 0, errors.Wrap(err, "resolve job ctx")
+		}
+		return id, err
+	}
+	if err := a.checkPermissions(ctx); err != nil {
+		return 0, errors.Wrap(err, "check permissions")
+	}
+	return pjsdb.JobID(jobID), err
+}
+
+func (a *apiServer) checkPermissions(ctx context.Context) error {
+	// Although there is a JOB resource, the JOB resource is scoped to specific job instances.
+	// The cluster resource is meant for cluster-wide permissions.
+	permissionResp, err := a.env.Permitter.GetPermissions(ctx, &auth.GetPermissionsRequest{
+		Resource: &auth.Resource{Type: auth.ResourceType_CLUSTER},
+	})
+	if err != nil && !errors.Is(err, auth.ErrNotActivated) {
+		return errors.Wrap(err, "get user permissions")
+	}
+	foundValidPermission := false
+	for _, p := range permissionResp.Permissions {
+		if p == auth.Permission_JOB_SKIP_CTX {
+			foundValidPermission = true
+			break
+		}
+	}
+	if !foundValidPermission {
+		return status.Errorf(codes.PermissionDenied, "insufficient privileges to omit the 'jobContext' field")
```

Comment:
Does jobContext need to be quoted? Do we use single quotes (`‘’`), double quotes (`“”`), foot/minute (`'`) or inch/second (`"`) markers in other similar error messages?

Reply:
Need is a _strong_ word, I think quotes help a reader quickly identify fields or extract values from a larger error message at the cost of some complexity when parsing. 

I'm swapping to double quotes as they seem like they're used in other error messages, especially formatted ones.